src/rules/order.js

```diff
@@ -781,6 +793,10 @@ module.exports = {
               'never',
             ],
           },
+          sortTypesAmongThemselves: {
```

Comment:
this is a bit wordy; let's rename it throughout to just `sortTypes`?

Reply:
I considered the shorter name at first, but I don't believe it's descriptive enough. The `import/order` rules already sort type-only imports; `"type"` is already a valid `groups` name, and its `groups` that actually controls if type-only imports are sorted or not sorted, right? With `sortTypes`, it wouldn't be immediately clear what turning off "sorting types" is supposed to mean with respect to the other rule options. You'd have scenarios where `sortTypes: false` but types are still being sorted...? And would `sortTypes: false` disable `named.types` sorting? This kind of ambiguity would confuse future me, at least, and would cause an unnecessary trip to the docs, docs which would have to become even more verbose to explain why `sortTypes` doesn't control if types are sorted.

Imo, `sortTypes` doesn't communicate the effect of flipping this switch and can even be confusing. We'd have `sortTypes`, `"type"` in `group`, `"type"` in `pathGroups.group`, `"type" ` in `pathGroupExcludedImportTypes` (which is already wordy and already somewhat-confusingly uses the word "Types" in its own name different to how the word is used in other option names like `named.types` and `newlines-between-types`), etc.

Whereas `sortTypesAmongThemselves` makes it clear that we're not interfering with the other "type"-related options, and we're not controlling if type-only imports are sorted vs not sorted but if they're sorted _among themselves_ vs if they're sorted _among the normals_.

Reply:
I think that works 😊